### DISCUSSION TITLE
Element-wise sum and product

### DISCUSSION THREAD

**Discussion by Ababwa** [bitshifter/glam-rs#462] [Ideas]
Use case for vectors:
Product for calculating volume.
Sum for getting taxicab distance.

Not sure what uses cases there might be for matrices, but seems apt to include them for completeness.

**Comment by bitshifter** [comment]
Happy to add product and sum for vectors, naming wise probably something like `product_elements` and `sum_elements` to differentiate from the existing `product` and `sum` implementations and to match other element wise methods like `max_element`. I wouldn't add these to matrices unless there's a concrete use case.

**Reply by Ababwa** [reply]
How about `element_product` and `element_sum`?

**Reply by bitshifter** [reply]
Sure, I like `sum_elements` but `product_elements` sounds weird.

**Comment by bitshifter** [comment]
Merged in #467
